app/src/main/java/com/github/polypoly/app/base/game/rules_and_lobby/GameLobbyActivity.kt

```diff
@@ -0,0 +1,193 @@
+                color = MaterialTheme.colors.background
+            ) {
+                Column(
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    Text("Game Lobby name: ${gameLobby.value.name}")
+                    Button(onClick = { /*TODO*/ }) {
+                        Text(text = "QUIT")
+                    }
+                    PlayerGrid()
+                    GoButton()
+                }
+            }
+        }
+
+    }
+
+    // ===================================================== GAME LOBBY COMPONENTS
+
+    @Composable
+    private fun PlayerGrid() {
+        val numberOfPlayers = gameLobby.value.usersRegistered.size
+        val numberOfRows: Int = numberOfPlayers / playerPerRow
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(Color.LightGray)
+                .height(400.dp),
+            horizontalArrangement = Arrangement.Center,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Column(
+                modifier = Modifier
+                    .padding(Padding.large)
+                    .fillMaxWidth(),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                for(i in 0 until numberOfRows) {
+                    Row {
+                        for (j in 0 until playerPerRow) {
+                            val index = i * playerPerRow + j
+                            if (index >= numberOfPlayers) {
+                                PlayerIcon(user = null)
+                            } else {
+                                PlayerIcon(user = gameLobby.value.usersRegistered[index])
+                            }
+                            if(j != playerPerRow - 1) {
+                                Spacer(modifier = Modifier.width(Padding.medium))
+                            }
+                        }
+                    }
+                    if(i != numberOfRows - 1) {
+                        Spacer(modifier = Modifier.height(Padding.medium))
+                    }
+                }
+            }
+        }
+    }
+
+    @Composable
+    private fun PlayerIcon(user: User?) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier
+                .background(Color.Gray)
+                .size(100.dp)
+        ) {
+            Box(
+                modifier = Modifier
+                    .background(
+                        color = if (user != null) MaterialTheme.colors.primary else MaterialTheme.colors.secondary,
+                        shape = CircleShape
+                    )
+                    .padding(20.dp)
+                    .size(30.dp)
+            )
+            if(user != null) {
+                Text(text = user.name)
+            }
+        }
+    }
+
+    /**
+     * TODO: dummy button that redirects to MapActivity
```

Comment:
I find the description of this TODO confusing because it sounds like a dummy button that redirects to MapActivity, while this is what the code below seems to do. What did you mean by this TODO?